Public/Src/Cache/ContentStore/Library/Tracing/EvictResult.cs

```diff
@@ -91,6 +92,23 @@ public EvictResult(ContentHashWithLastAccessTimeAndReplicaCount contentHash, lon
         /// </summary>
         public DateTime CreationTime { get; }
 
+        /// <summary>
+        ///     Convert to the <see cref="DeleteResult"/> class.
+        /// </summary>
+        public DeleteResult ToDeleteResult(ContentHash contentHash)
+        {

```

Comment:
I would argue that this conversion method more naturally belongs on DeleteResult, because DeleteResult depends on EvictResult, not vice-versa.

Reply:
I would agree, but `DeleteResult` must be defined in Interfaces with `IContentStore`, while `EvictResult` currently lives in Library. It seems dangerous to move `EvictResult` from Library to Interfaces, but I can give it a try if needed.

Another option would be to move this converter into its own extension method in Library, but I think that's even more messy than defining it in `EvictResult`. Thoughts?

---
In reply to: [299291008](https://github.com/microsoft/BuildXL/pull/543#discussion_r299291008) [](ancestors = 299291008)